aiida/cmdline/commands/cmd_computer.py

```diff
@@ -161,6 +162,55 @@ def _computer_create_temp_file(transport, scheduler, authinfo):  # pylint: disab
     return True, None
 
 
+def time_use_login_shell(authinfo, auth_params, use_login_shell: bool) -> float:
+    """Execute the ``whoami`` over the transport for the given ``use_login_shell`` and report the time taken."""
+    import time
+
+    auth_params['use_login_shell'] = use_login_shell
+    authinfo.set_auth_params(auth_params)
+
+    time_start = time.time()
+
+    with authinfo.get_transport() as transport:
+        transport.whoami()
+
+    return time.time() - time_start
+
+
+def _computer_use_login_shell_performance(transport, scheduler, authinfo):  # pylint: disable=unused-argument
+    """Execute a command over the transport with and without the ``use_login_shell`` option enabled.
+
+    By default, a computer is configured to use a login shell when connecting to it using the transport. For certain
+    environments, loading the login script of the shell is quite costly which will significantly slow down all commands
+    executed. This has a big impact on the throughput of calculation jobs. This test tries to execute a simple command
+    both with and without using a login shell and emits a warning if the ratio of the execution time exceeds 2.
+    """
+    factor = 2
+
+    auth_params = authinfo.get_auth_params()
+    auth_params_clone = deepcopy(auth_params)
+
+    # If ``use_login_shell=False`` we don't need to test for it being slower.
+    if not auth_params.get('use_login_shell', True):
+        return True, None
+
+    try:
+        timing_false = time_use_login_shell(authinfo, auth_params_clone, False)
+        timing_true = time_use_login_shell(authinfo, auth_params_clone, True)
+    finally:
+        authinfo.set_auth_params(auth_params)
+
+    if timing_true > (timing_false * factor):
```

Comment:
Done. But I have questions about the implementation. My first instinct was to use `math.isclose` however this does not work as I expected. If I set relative tolerance of 0.5 (factor of two), it would still mark two very small numbers, say 0.01 and 0.011, as close even for a negligible absolute tolerance of `0.000000001`. The same goes for `numpy.isclose`. They effectively consider both tolerances, and as long as one is respected, the numbers are considered close. I think in this case, we want it to fail as soon as either difference (abs or relative) exceeds the tolerance, by itself. Is this correct?

Reply:
The idea of the absolute threshold was to avoid false positives when the absolute time delta is very small and could be dominated by "noise".
So I would have raised the warning only if both thresholds are exceeded.

Come to think of it, I guess we mostly care about absolute time differences here anyhow, so as long as we have a somewhat reliable timing (via the average of the 3 attempts), one could also go with an absolute time threshold (as in the `verdi` load tests).

Reply:
Ok, simplified to just checking the difference is within 100 ms